flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java

```diff
@@ -387,11 +387,11 @@ public void release(MemorySegment segment) {
 	 *
 	 * @param segments The segments to be released.
 	 * @throws NullPointerException Thrown, if the given collection is null.
-	 * @throws IllegalArgumentException Thrown, id the segments are of an incompatible type.
+	 * @throws IllegalArgumentException Thrown, if the segments are of an incompatible type.
 	 */
 	public void release(Collection<MemorySegment> segments) {
 		if (segments == null) {
-			return;
+			throw new NullPointerException();
```

Comment:
If we  get a `null` to release, why can't we just skip it? I think it's a defensive code not a bug. Or can you post some bad situation about this?

Reply:
> If we get a `null` to release, why can't we just skip it? I think it's a defensive code not a bug. Or can you post some bad situation about this?

Hi Aitozi, thanks a lot for your comment. The JavaDoc for this method declares "@throws NullPointerException Thrown, if the given collection is null." So we should either throw an NPE, or change the JavaDoc, to make it consistent. 
By referencing other methods in this class, I think it is preferable to make the method throw NPE, because the methods should follow a consistent style.

Reply:
It's truly inconsistent, but this fix will change the default behaviour of the release operation, and the `release(MemorySegment segment)` also skip the null...  I think may be it need @StephanEwen author of the `MemoryManager.java` to confirm, Thanks.

Reply:
That sounds reasonable. Thank you for your comments. Hi @StephanEwen, would you please give some comments?

Reply:
Unless there is a real bug caused by this in some setup, I would not change this.
Such changes often have unforeseen side effects, so I would avoid doing them for "looks better" or "looks more correct" reasons.

Reply:
Let's change the JavaDocs instead to match the implementation.

Reply:
@StephanEwen Sounds reasonable. I have revised the PR accordingly. Please take a look.